### PR TITLE
Milestone2

### PR DESCRIPTION
**Pull request by alekott**

### CODE REVIEW

File: src/Shop.java [hunk 1]
```diff
@@ -51,14 +80,72 @@ public ProductDetails findProductDetailsByProduct(Product productToFind) {
...
             products.add(newProductDetails);
         } else {
             findByProduct.quantity += quantityToAdd;
         }
     }
-    
+
+    public Product getProduct(int productId){
+        for (ProductDetails productDetails : products) {
+            if ( productDetails.product.id == productId){
```
Comment (yuriyyakym) [inline src/Shop.java]: Inconsistent space after a bracket.

File: src/Shop.java [hunk 2]
```diff
@@ -51,14 +80,72 @@ public ProductDetails findProductDetailsByProduct(Product productToFind) {
...
+    public void addService(Service service) {
+        this.services.add(service);
+    }
+
+    public void deleteService(Service serviceToDelete) {
+        this.services.remove(serviceToDelete);
+    }
+
+    public Service getServiceById(int serviceId){
+        for ( Service service : services){
```
Comment (yuriyyakym) [inline src/Shop.java]: Inconsistent space after bracket.

File: src/ShoppingCenter.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        for( Shop shop : shops){
+            if(shop.id == shopId){
+                return shop;
+            }
+        }
+        return null;
+    }
+    */
+
+    public void updateShop(int shopId, Shop modifiedShop){
```
Comment (yuriyyakym) [inline src/ShoppingCenter.java]: No space before curly bracket

File: src/Shop.java [hunk 4]
```diff
@@ -51,14 +80,72 @@ public ProductDetails findProductDetailsByProduct(Product productToFind) {
...
+            if (service.id == serviceId){
+                return service;
+            }
+        }
+        return null;
+    }
+
+    public void updateService(int serviceId, Service modifiedService){
```
Comment (yuriyyakym) [inline src/Shop.java]: Same here

File: src/Shop.java [hunk 5]
```diff
@@ -51,14 +80,72 @@ public ProductDetails findProductDetailsByProduct(Product productToFind) {
...
+
+    public void addService(Service service) {
+        this.services.add(service);
+    }
+
+    public void deleteService(Service serviceToDelete) {
+        this.services.remove(serviceToDelete);
+    }
+
+    public Service getServiceById(int serviceId){
```
Comment (yuriyyakym) [inline src/Shop.java]: No space before curly bracket.

File: src/ShoppingCenter.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        deleteShop(shopId);
+        modifiedShop.id = shopId;
+        addShop(modifiedShop);
+    }
+
+    public void deleteShop (int shopId){
```
Comment (yuriyyakym) [inline src/ShoppingCenter.java]: same here

Comment (yuriyyakym) [reply src/ShoppingCenter.java]: just fix it everywhere

File: src/ShoppingCenter.java [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+        for (Product product : list){
+               if (!typesList.contains(product.type)){
+                   typesList.add(product.type);
+            }
+        }
+        return typesList;
+    }
+
+    public List<String> getAllProductTypes2(){
+        return getAllProducts().stream()
```
Comment (yuriyyakym) [inline src/ShoppingCenter.java]: Just make items of that stream unique (use `.distinct` method)

File: src/Product.java [hunk 8]
```diff
@@ -1,14 +1,18 @@
 public class Product {
 
+    private static  int nextId = 0;
+
+    public int id;
     private String name;
     private double price;
-    private String type;
+    public String type;
```
Comment (yuriyyakym) [inline src/Product.java]: Keep public fields together.

File: src/Shop.java [hunk 9]
```diff
@@ -1,19 +1,39 @@
...
-    private List<Service> services;
+    private static int nextId = 0;
+
+    public int id;
+    public String name;
+    private Location location;
+    public List<ProductDetails> products;
+    public List<String> types;
+    public List<Service> services;
     private List<Employee> employees;
```
Comment (yuriyyakym) [inline src/Shop.java:14]: keep private fields together

**Review by yuriyyakym — CHANGES_REQUESTED** [review]
Nice code. Mostly styleguide changes are required.

File: src/Shop.java [hunk 10]
```diff
@@ -1,22 +1,42 @@
...
+        this.services = new LinkedList<Service>();
+        this.employees = new LinkedList<Employee>();
     }
 
-    public void hireEmployee(Employee potentialEmployee) {
+    public Shop(String name, int floor, int box) {                  //method overloading - przeladowanie metody
+        this(name, floor, box, new LinkedList<String>());
+    }
+
+
```
Comment (yuriyyakym) [inline src/Shop.java:31]: Redundant empty line

File: src/ShoppingCenter.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+                .collect(Collectors.toList());
+    }
+
+    public Shop getShop(int shopId){
+        return shops.stream()
+                .filter(shop -> shop.id == shopId)
+                .findAny()
+                .orElse(null);
+    }
+
```
Comment (yuriyyakym) [inline src/ShoppingCenter.java:82]: redundant empty line

**Review by michal-marciniec — APPROVED** [review]

**Pull request merged**